src/context.tsx

```diff
@@ -56,107 +56,25 @@ const initial_filters: Filters = getDefaultFilters();
-    if ((filters[filter_key as FilterType] instanceof Set) && (filters[filter_key as FilterType].size > 0)) {
-      if (record[filter_key] === null) {
-        return false;
-      }
-      // Handle case where field to be filtered is an array
-      if (Array.isArray(record[filter_key])) {
-        // Note: isotypes filter works on an 'and' basis
-        // Unlike other filters, which work on an 'or' basis
-        // TODO: make logic flow here more generalized in case we need other filters in the future with similar behaviour
-        if (filter_key === 'isotypes_reported') {
-          let match = true;
-          filters[filter_key].forEach((item: string) => {
-            if (!(record[filter_key].includes(item))) {
-              match = false;
-            }
-          });
-          return match;
-        }
-
-        if (filter_key === 'publish_date') {
-          const publishDate = record[filter_key];
-          const dateInMillis = publishDate instanceof Array ? Date.parse(publishDate[0] as string) : Date.parse(publishDate as string)
-          const dates: number[] = Array.from(filters[filter_key].values())
-          return dateInMillis >= dates[0] && dateInMillis <= dates[1]
-        }
-        // Iterate through the record's values and check if any of them
-        // match the values accepted by the filter
-        let in_filter = false;
-        for (let i = 0; i < record[filter_key].length; i++) {
-          if (filters[filter_key as FilterType].has(record[filter_key][i])) {
-            in_filter = true;
-            break;
-          }
-        }
-        if (!in_filter) {
-          return false;
-        }
-      }
-      else {
-        if (!(filters[filter_key as FilterType].has(record[filter_key]))) {
-          return false;
-        }
-      }
-    }
-  }) 
-  return !array.some(o => o === false);
-}
-
-// export function filterRecords(filters: Filters, records: AirtableRecord[]) {
-//   const filter_function = buildFilterFunction(filters);
-//   if (records) {
-//     const filtered_records = records.filter(filter_function);
-//     return filtered_records
-//   }
-//   return [];
-// }
-
-function recomputeFilterOptions(records: AirtableRecord[], all_filter_options: Filters, filters: Filters) {
-  const options = getFilterOptions(records);
-  for (const filter in filters) {
-    // If the filter is already in use, show all filter options
-    // Since options within the same filter work on an "or basis"
-    // (e.g. risk_of_bias: ["Low", "Medium"] means include records with Low or Medium bias)
-    // the user will never get 0 records back if they choose more filter options for a filter
-    // that they've already selected an option for
-    if (filters[filter as FilterType].size > 0) {
-      options[filter as FilterType] = all_filter_options[filter as FilterType];
-    }
-  }
-  return options;
-}
-
 function getFilterOptions(records: AirtableRecord[]) {
```

Comment:
Again comment a TODO here to delete this function when we integrate the new backend endpoint